src/dblib_stubs.c

```diff
@@ -389,7 +389,7 @@ CAMLexport value ocaml_freetds_dbnextrow(value vdbproc)
       data = dbdata(dbproc, c); /* pointer to the data, no copy! */
       len = dbdatlen(dbproc, c); /* length, in bytes, of the data for
                                     a column. */
-      if (len == 0 || data == NULL) {
+      if (data == NULL) {
```

Comment:
http://infocenter.sybase.com/help/index.jsp?topic=/com.sybase.help.ocs_12.5.1.dblib/html/dblib/X69467.htm

> Returns
> A BYTE pointer to the data for the particular column of interest. Be sure to cast this pointer into the proper type. A NULL BYTE pointer is returned if there is no such column or if the data has a null value. To make sure that the data is really a null value, you should always check for a return of 0 from dbdatlen.

So technically we should distinguish `data == NULL && len == 0` -> NULL and `data == NULL && len < 0` -> column doesn't exist.